src/client/errors.ts

```diff
@@ -0,0 +1,57 @@
+export enum OpenError {
+    AppNotFound = 'AppNotFound',
+    ErrorOnLaunch = 'ErrorOnLaunch',
+    AppTimeout = 'AppTimeout',
+    ResolverUnavailable = 'ResolverUnavailable'
+}
+
+export enum ResolveError {
+    NoAppsFound = 'NoAppsFound',
+    ResolverUnavailable = 'ResolverUnavailable',
+    ResolverTimeout = 'ResolverTimeout',
+    InvalidContext = 'InvalidContext'
+}
+
+export class FDC3Error extends Error {
+    /**
+     * If error is FDC3 specific, serialize it as { code, message } so it can be identified as an `FDC3Error` at the client's end.
+     * Otherwise return the error itself
+     * @param error The error
+     */
+    public static serialize(error: Error | FDC3Error): Error {
+        // TODO: is use of `instanceof` OK? or should use `if (error && error.code && error.message)` ?
```

Comment:
Are you able to make a call on this and get rid of the `TODO`? `instanceof` _should_ be fine right? Do we expect this to be called with anything that's come across the wire but hasn't been properly deserialized?

Reply:
Yes! Actually that (and a bunch more TODO's) are addressed in the [second part of the story](https://github.com/spcfran/fdc3-service/pull/1/commits/e31ddb8be86639d8322c6171166add5b55991688)

@pjbroadbent and I agreed `instanceof` is ok

I plan to submit that PR as soon as this one is merged into `feature`, so I guess it's OK to skip it here for now?